src/dialog/provider.js

```diff
@@ -127,31 +92,17 @@ export default class FlowProvider extends EventEmitter2 {
 
     // What goes in the ui.json file
     const uiContent = {
-      nodes: flow.nodes.map(node => ({
-        id: node.id,
-        position: { x: node.x, y: node.y }
-      })),
+      nodes: flow.nodes.map(node => ({ id: node.id, position: { x: node.x, y: node.y } })),
       links: flow.links
     }
 
     // What goes in the .flow.json file
     const flowContent = {
-      version: flow.version,
-      startNode: flow.startNode,
-      catchAll: flow.catchAll,
-      nodes: flow.nodes,
-      skillData: flow.skillData
+      ..._.pick(flow, 'version', 'catchAll', 'startNode', 'skillData'),
+      nodes: flow.nodes.map(node => _.omit(node, 'x', 'y', 'lastModified'))
     }
 
-    flowContent.nodes.forEach(node => {
-      // We remove properties that don't belong in the .flow.json file
-      delete node['x']
-      delete node['y']
-      delete node['lastModified']
-    })
-
-    const relDir = this.botfile.flowsDir || './flows'
-    const flowPath = path.resolve(this.projectLocation, relDir, './' + flow.location)
+    const flowPath = flow.location
     const uiPath = flowPath.replace(/\.flow\.json/i, '.ui.json')
```

Comment:
can go to a utility method, repeating the code from above

Reply:
Done.